packages/faucet/src/addresses.ts

```diff
@@ -1,12 +1,14 @@
 import { fromBech32 } from "@cosmjs/encoding";
 
+/**
+ * Check length based on Bech32 from {@link https://github.com/bitcoin/bips/blob/e1e7b77c027b3d40d07d306cc75c2b5859c91db2/bip-0173.mediawiki#bech32 | BIP173}
+ */
 export function isValidAddress(input: string, requiredPrefix: string): boolean {
   try {
     const { prefix, data } = fromBech32(input);
-    if (prefix !== requiredPrefix) {
-      return false;
-    }
-    return data.length === 20;
```

Comment:
I think the input length should be checked in fromBech32 already. Let's just fix this by checking the data range

```ts
return data.length >= 20 && data.length <= 32;
```